src/Valit/ValitRules.cs

```diff
@@ -60,6 +60,16 @@ IValitRules<TObject> IValitRules<TObject>.Ensure<TProperty>(Func<TObject, TPrope
             return this;
         }
 
+        IValitRules<TObject> IValitRules<TObject>.Ensure<TProperty>(Func<TObject, TProperty> selector, Func<IValitRule<TObject, TProperty>, TObject, IValitRule<TObject, TProperty>> ruleFunc)
+        {                       
+            selector.ThrowIfNull();
+            ruleFunc.ThrowIfNull();
+
+            var lastEnsureRule = new ValitRule<TObject, TProperty>(selector, _messageProvider, ruleFunc);
```

Comment:
You should call ``AddEnsureRulesAccessors`` here to get all rules from the chain. BTW this name is stupid :D  We should change that to ``AddEnsureRules`` instead.

Reply:
I can't call `AddEnsureRuleAccessors` because it has different arguments. I will do some change to get all rules from the chain.

Reply:
sorry, my bad :D